From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 29 Oct 2018 02:30:21 +0000
Subject: netdev: Hide netdev_notifier_info_ext from modules
Forwarded: not-needed

Since commit 8d59c3a6376b "net: ipv4: update fnhe_pmtu when first
hop's MTU changes", the NETDEV_CHANGEMTU notification is now sent with
a parameter type of struct netdev_notifier_info_ext, an extension of
struct netdev_notifier_info.  However, a module that relies on this
could be loaded against an older kernel version that doesn't set the
extra members. Make sure modules don't use this structure until the
next ABI bump.

---
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -2338,12 +2338,14 @@ struct netdev_notifier_info {
 	struct net_device *dev;
 };
 
+#ifndef MODULE
 struct netdev_notifier_info_ext {
 	struct netdev_notifier_info info; /* must be first */
 	union {
 		u32 mtu;
 	} ext;
 };
+#endif
 
 struct netdev_notifier_change_info {
 	struct netdev_notifier_info info; /* must be first */
